CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 Hennadii Stepanov
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+cmake_minimum_required(VERSION 3.1)
+if(CMAKE_VERSION VERSION_GREATER 3.14)
+  # MSVC warning flags are not in CMAKE_<LANG>_FLAGS by default.
+  # See: https://cmake.org/cmake/help/latest/policy/CMP0092.html
+  cmake_policy(SET CMP0092 NEW)
+endif()
+project(secp256k1 VERSION 0.1.0 LANGUAGES C)
+
+set(CMAKE_C_STANDARD 90)
+set(CMAKE_C_EXTENSIONS OFF)
+
+option(ENABLE_DEV_MODE "enable all binaries and modules by default but individual options can still be overridden explicitly" OFF)
+
+option(BUILD_BENCHMARK "build benchmarks" ON)
+option(BUILD_TESTS "build tests" ON)
+option(BUILD_EXHAUSTIVE_TESTS "build exhaustive tests" ON)
+option(BUILD_EXAMPLES "build examples" ${ENABLE_DEV_MODE})
+
+option(ENABLE_MODULE_ECDH "enable ECDH module" ${ENABLE_DEV_MODE})
+option(ENABLE_MODULE_RECOVERY "enable ECDSA pubkey recovery module" ${ENABLE_DEV_MODE})
+option(ENABLE_MODULE_EXTRAKEYS "enable extrakeys module" ${ENABLE_DEV_MODE})
+option(ENABLE_MODULE_SCHNORRSIG "enable schnorrsig module" ${ENABLE_DEV_MODE})
+
+option(ALLOW_EXPERIMENTAL "allow experimental configure options" ${ENABLE_DEV_MODE})
+option(USE_EXTERNAL_DEFAULT_CALLBACKS "enable external default callback functions" OFF)
+option(COVERAGE "enable compiler flags to support coverage analysis" OFF)
+
+set(ECMULT_WINDOW_SIZE "auto" CACHE STRING "window size for ecmult precomputation for verification, specified as integer in range [2..24]. \"auto\" is a reasonable setting for desktop machines (currently 15). [default=auto]")
+if(ECMULT_WINDOW_SIZE STREQUAL auto)
+  set(ECMULT_WINDOW_SIZE 15)
+endif()
+if(NOT ECMULT_WINDOW_SIZE MATCHES ^[1-9][0-9]*$ OR ECMULT_WINDOW_SIZE LESS 2 OR ECMULT_WINDOW_SIZE GREATER 24)
+  message(FATAL_ERROR "ECMULT_WINDOW_SIZE value is \"${ECMULT_WINDOW_SIZE}\", but must an integer in range [2..24] or \"auto\".")
+endif()
+
+set(ECMULT_GEN_PREC_BITS "auto" CACHE STRING "Precision bits to tune the precomputed table size for signing, specified as integer 2, 4 or 8. \"auto\" is a reasonable setting for desktop machines (currently 4). [default=auto]")
```

Comment:
```suggestion
set(ECMULT_GEN_PREC_BITS "auto" CACHE STRING "Precision bits to tune the precomputed table size for signing, specified as integer 2, 4 or 8. \"auto\" is a reasonable setting for desktop machines (currently 4). [default=auto]")
set_property(CACHE ECMULT_GEN_PREC_BITS PROPERTY STRINGS "auto" 2 4 8)
```

Reply:
This makes `ccmake` and `cmake-gui` offer a nicer selection. We should probably do the same above. Listing the numbers from 2 to 24 looks strange but why not. (We'd like to restrict this in the future anyway.)